pkger/http_server_template.go

```diff
@@ -332,6 +333,40 @@ func (s *HTTPServerTemplates) apply(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Reject use of server-side jsonnet with /api/v2/templates/apply
+	if encoding == EncodingJsonnet {
+		s.api.Err(w, r, &errors.Error{
+			Code: errors.EUnprocessableEntity,
+			Msg:  fmt.Sprintf("template from source(s) had an issue: %s", ErrInvalidEncoding.Error()),
+		})
+		return
+	}
+
+	var remotes []string
+	for _, rem := range reqBody.Remotes {
+		remotes = append(remotes, rem.URL)
+	}
+	remotes = append(remotes, reqBody.RawTemplate.Sources...)
+
+	for _, rem := range remotes {
+		// While things like '.%6Aonnet' evaluate to the default encoding (yaml), let's unescape and catch those too
+		decoded, err := url.QueryUnescape(rem)
```

Comment:
We don't want to return the error here?

Reply:
Or does that let too much security information escape?

Reply:
There may be some security concern, although there shouldn't be much of an issue since our `api.Err(...)` methods return sanitized errors to clients. Mostly I am keeping these consistent with the IDPE error messages.

Reply:
Hmm, I see @jdstrand has already merged the code in IDPE, so probably we shouldn't become incompatible.

Reply:
If you're referring to the specific error from `url.QueryUnescape()` instead of a generic error, I erred on the side of caution since I felt legitimate use of the API should not be causing this error and so not giving the attacker a generic error was fine. If we find this to be problematic in practice, we can return a more specific error.